### PR TITLE
planning: add a stop timeout on static pedestrian/bicycles on crosswalk

### PR DESCRIPTION
**Pull request by jmtao**

### CODE REVIEW

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 1]
```diff
@@ -184,11 +212,34 @@ void Crosswalk::MakeDecisions(Frame* const frame,
...
       }
 
+      if (stop) {
+        const double kMaxStopSpeed = 0.3;
+        auto obstacle_speed = std::hypot(perception_obstacle.velocity().x(),
+                                         perception_obstacle.velocity().y());
+        if (stopped_at_crosswalk && obstacle_speed <= kMaxStopSpeed) {
+          // check on stop timer
+          if (crosswalk_stop_timer.first == crosswalk_overlap->object_id) {
+            auto stop_timers = crosswalk_stop_timer.second;
```
Comment (startcode) [inline modules/planning/tasks/traffic_decider/crosswalk.cc]: use `const auto& stop_timers`

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 2]
```diff
@@ -210,10 +261,16 @@ void Crosswalk::MakeDecisions(Frame* const frame,
...
 
   for (auto crosswalk_to_stop : crosswalks_to_stop) {
+    // build stop decision
     BuildStopDecision(frame, reference_line_info,
                       const_cast<hdmap::PathOverlap*>(crosswalk_to_stop.first),
                       crosswalk_to_stop.second);
   }
+
+  // update CrosswalkStatus
+  GetPlanningStatus()->clear_crosswalk();
```
Comment (startcode) [inline modules/planning/tasks/traffic_decider/crosswalk.cc]: do you want to reset PlanningStatus::crosswalk()?

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 3]
```diff
@@ -86,6 +97,23 @@ void Crosswalk::MakeDecisions(Frame* const frame,
       continue;
     }
 
+    bool stopped_at_crosswalk = false;
+    CrosswalkStopTimer crosswalk_stop_timer;
+    if (CheckADCkStop(reference_line_info,
+                      const_cast<hdmap::PathOverlap*>(crosswalk_overlap))) {
+      stopped_at_crosswalk = true;
+
+      CrosswalkStatus crosswalk_status = GetPlanningStatus()->crosswalk();
```
[the remark this answers is not in the document]
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: it is not a const.  it's a local variable and can be updated.

File: modules/planning/tasks/traffic_decider/crosswalk.cc (hunk 1 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: it is not a const.  it's a local variable and can be updated.

File: modules/planning/tasks/traffic_decider/crosswalk.cc (hunk 2 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: yes. we'll update crosswalk status. next line with TODO

File: modules/planning/proto/planning_status.proto [hunk 4]
```diff
@@ -24,6 +24,18 @@ message ChangeLaneStatus {
   optional double timestamp = 3;
 }
 
+message StopTimer {
+  required string obstacle_id = 1;
```
[the remark this answers is not in the document]
Comment (jmtao) [reply modules/planning/proto/planning_status.proto]: fixed

File: modules/planning/proto/planning_status.proto [hunk 5]
```diff
@@ -24,6 +24,18 @@ message ChangeLaneStatus {
...
+  // the timestamp when start stopping for the crosswalk
+  optional double stop_time = 2;
+}
+
+message CrosswalkStatus {
+  required string crosswalk_id = 1;
```
[the remark this answers is not in the document]
Comment (jmtao) [reply modules/planning/proto/planning_status.proto]: fixed

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 6]
```diff
@@ -228,6 +285,38 @@ bool Crosswalk::FindCrosswalks(ReferenceLineInfo* const reference_line_info) {
   return crosswalk_overlaps_.size() > 0;
 }
 
+/**
+ * @brief: check valid stop_sign stop
+ */
+bool Crosswalk::CheckADCkStop(ReferenceLineInfo* const reference_line_info,
+                             hdmap::PathOverlap* const crosswalk_overlap) {
```
[the remark this answers is not in the document]
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: fixed

**Pull request merged**